test/gateway/eip20_cogateway/confirm_revert_stake_intent.js

```diff
@@ -0,0 +1,308 @@
+const StubData = require('../../data/stake_revoked_1.json');
+
+let MessageStatusEnum = messageBus.MessageStatusEnum;
+
+contract('EIP20CoGateway.confirmRevertStakeIntent() ', function (accounts) {
+  let utilityToken,
+    coGateway,
+    owner = accounts[0], messageHash, amount,
+    stakeRequest = StubData.gateway.stake.params;
+
+  beforeEach(async function () {
+
+    let organization = accounts[2];
+
+    utilityToken = await UtilityToken.new(
+      accounts[9],
+      "",
+      "",
+      18,
+      organization, //organization,
+      {from: owner},
+    );
+
+    coGateway = await EIP20CoGateway.new(
+      StubData.contracts.mockToken,
+      utilityToken.address,
+      accounts[6], //dummyStateRootProvider
+      StubData.co_gateway.constructor.bounty,
+      organization,
+      StubData.contracts.gateway,
+      StubData.co_gateway.constructor.burner,
+    );
+
+    amount = new BN(stakeRequest.amount, 16);
+
+    let intentHash = coGatewayUtils.hashStakeIntent(
+      amount,
+      stakeRequest.beneficiary,
+      StubData.contracts.gateway,
+    );
+
+    await utilityToken.setCoGatewayAddress(coGateway.address);
+
+    messageHash = await coGateway.setMessage.call(
+      intentHash,
+      new BN(stakeRequest.nonce, 16),
+      new BN(stakeRequest.gasPrice, 16),
+      new BN(stakeRequest.gasLimit, 16),
+      stakeRequest.staker,
+      stakeRequest.hashLock,
+    );
+
+    await coGateway.setMessage(
+      intentHash,
+      new BN(stakeRequest.nonce, 16),
+      new BN(stakeRequest.gasPrice, 16),
+      new BN(stakeRequest.gasLimit, 16),
+      stakeRequest.staker,
+      stakeRequest.hashLock,
+    );
+
+    await coGateway.setMints(messageHash, stakeRequest.beneficiary, amount);
+
+  });
+
+  it('should confirm revert stake intent', async function () {
+
+    await coGateway.setInboxStatus(messageHash, MessageStatusEnum.Declared);
+
+    let blockHeight = new BN(StubData.gateway.revert_stake.proof_data.block_number, 16);
+    let storageRoot = StubData.gateway.revert_stake.proof_data.storageHash;
+
+    let storageProof = StubData.gateway.revert_stake.proof_data.storageProof[0].serializedProof;
+    await coGateway.setStorageRoot(blockHeight, storageRoot);
+
+    let tx = await coGateway.confirmRevertStakeIntent(
+      messageHash, blockHeight, storageProof
+    );
+
+    assert.strictEqual(
+      tx.receipt.status,
+      true,
+      'Transaction should success.',
+    );
+
+  });
+
+  it('should emit RevertStakeIntentConfirmed event', async function () {
+
+    await coGateway.setInboxStatus(messageHash, MessageStatusEnum.Declared);
+
+    let blockHeight = new BN(StubData.gateway.revert_stake.proof_data.block_number, 16);
+    let storageRoot = StubData.gateway.revert_stake.proof_data.storageHash;
+
+    let storageProof = StubData.gateway.revert_stake.proof_data.storageProof[0].serializedProof;
+    await coGateway.setStorageRoot(blockHeight, storageRoot);
```

Comment:
Following code is used in multiple tests, may be we can move this to a function.
```
    let blockHeight = new BN(StubData.gateway.revert_stake.proof_data.block_number, 16);
    let storageRoot = StubData.gateway.revert_stake.proof_data.storageHash;

     let storageProof = StubData.gateway.revert_stake.proof_data.storageProof[0].serializedProof;
    await coGateway.setStorageRoot(blockHeight, storageRoot);
```

Reply:
Done